src/Microsoft.AspNet.Routing/Template/TemplateBinder.cs

```diff
@@ -471,13 +472,24 @@ public bool Accept(string value)
                 BufferState = SegmentState.Inside;
                 UriState = SegmentState.Inside;
 
-                _uri.Append(value);
+                // Allow the first segment to have a leading slash.
+                // This prevents the leading slash from PathString segments from being encoded.
+                if (_uri.Length == 0 && value.Length > 0 && value[0] == '/')
```

Comment:
Yeah, I see. I'm not super worried about this check, really you could just **never** encode `/` without the check. We've had a long-standing bug about not encoding `/` inside of segments, and we've decided multiple times not to address it.